src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -166,29 +165,18 @@ private static Task<Document> UpdateDocumentAsync(Document document, SyntaxNode
                     foreach (var contentPart in contentParts)
                     {
                         currentContentIsStringOrCharacterLiteral = syntaxFacts.IsInterpolatedStringText(contentPart);
-                        if (currentContentIsStringOrCharacterLiteral)
+                        if (currentContentIsStringOrCharacterLiteral && previousContentWasStringLiteralExpression)
                         {
-                            // if the piece starts with a text and the previous part was a string, merge the two parts (see also above)
-                            if (previousContentWasStringLiteralExpression)
-                            {
-                                var contentTextToken = contentPart.GetFirstToken();
-                                var newText = ConcatinateTextToTextNode(generator, content.Last(), contentTextToken.Text);
-                                content[^1] = newText;
-                            }
-                            else
-                            {
-                                content.Add(contentPart);
-                            }
-                        }
-                        else if (syntaxFacts.IsInterpolation(contentPart))
-                        {
-                            content.Add(contentPart);
-                            previousContentWasStringLiteralExpression = false;
+                            // if piece starts with a text and the previous part was a string, merge the two parts (see also above)
+                            var newText = ConcatinateTextToTextNode(generator, content.Last(), contentPart.GetFirstToken().Text);
```

Comment:
1. spelling: "Concatinate"
2. consider giving an example of what this means.

Reply:
Done in 6c75e923 and 2772cef7